solutions/usaco-988.mdx

```diff
@@ -112,4 +115,65 @@ print(ans, file=open('photo.out', 'w'))
 ```
 
 </PySection>
+<JavaSection>
+	
+**Time Complexity:** $\mathcal{O}(N^2)$
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class Photoshoot {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("photo");
+
+		int N = io.nextInt();
+		int[] bessieList = new int[N];
+		int[] correctList = new int[N];
+		// Used to prevent repeated values in our permutation
+		Set<Integer> isItThere = new HashSet<>();
+
+		for (int i = 1; i < N; i++) {
+			bessieList[i] = io.nextInt();
+		}
+
+		// Iterate through every possible starting cow and see if we can
+		// construct a permutation based on that.
+		for (int i = 1; i <= N; i++) {
+			correctList[0] = i;
+			isItThere = new HashSet<>();
+			for (int j = 1; j < N; j++) {
+				correctList[j] = bessieList[j] - correctList[j - 1];
+			}
+			boolean bad = false;
+			for (int k = 0; k < N; k++) {
+				// Our permutation isn't valid because there are unacceptable values
+				// So, we can stop searching. 
+				if (isItThere.contains(correctList[k]) || correctList[k] > N 
+					|| correctList[k] <= 0) {
+					bad = true;
+					break;
+				}
+				isItThere.add(correctList[k]);
+			}
+			// We've found a valid permutation since we started from 1, the 
+			// first one we find will be the lexicographically smallest one
+			if (!bad) {
+				for (int j = 0; j < N; j++) {
+					io.print(correctList[j]);
+					if (j < N - 1) {
+						io.print(" ");
+					}
+				}
+				io.print("\n");
+				io.close();
+			}
+		}
+	}
+	// CodeSnip{Kattio}
```

Comment:
remove this comment (unless it's needed, idk what it does, i just assumed u forgot to remove)

Reply:
it adds the collapsible kattio snipppet